packages/application-generic/src/services/filters/filter-conditions.service.ts

```diff
@@ -0,0 +1,636 @@
+      passedFilters: [],
+    });
+  }
+
+  private sumFilters(summary: IUsedFilters, condition: ICondition) {
+    let type: string = condition.filter?.toLowerCase();
+
+    if (
+      condition.filter === FILTER_TO_LABEL.isOnline ||
+      condition.filter === FILTER_TO_LABEL.isOnlineInLast
+    ) {
+      type = 'online';
+    }
+
+    if (!type) {
+      type = condition.filter;
+    }
+
+    type = type?.toLowerCase();
+
+    if (condition.passed && !summary.passedFilters.includes(type)) {
+      summary.passedFilters.push(type);
+    }
+
+    if (!condition.passed && !summary.failedFilters.includes(type)) {
+      summary.failedFilters.push(type);
+    }
+
+    if (!summary.filters.includes(type)) {
+      summary.filters.push(type);
+    }
+
+    return summary;
+  }
+
+  private async handleGroupFilters(
+    filter: StepFilter,
+    filterProcessingDetails: FilterProcessingDetails,
+    filterData: IFilterData,
+    variables: IFilterVariables,
+    properties: IFilterProperties
+  ): Promise<boolean> {
+    if (
+      ![FieldLogicalOperatorEnum.AND, FieldLogicalOperatorEnum.OR].includes(
+        filter.value
+      )
+    ) {
+      return false;
+    }
+
+    const matchedOtherFilters = await this.filterAsync(filter.children, (i) =>
+      this.processFilter(
+        i,
+        filterProcessingDetails,
+        filterData,
+        variables,
+        properties
+      )
+    );
+
+    if (filter.value === FieldLogicalOperatorEnum.OR) {
+      return !!matchedOtherFilters;
+    }
+
+    if (filter.value === FieldLogicalOperatorEnum.AND) {
+      return filter.children.length === matchedOtherFilters.length;
+    }
+
+    return false;
+  }
+
+  private async processFilter(
+    filter: FilterParts,
+    filterProcessingDetails: FilterProcessingDetails,
+    filterData: IFilterData,
+    variables: IFilterVariables,
+    properties: IFilterProperties
+  ): Promise<boolean> {
+    let passed = false;
+
+    if (filter.on === FilterPartTypeEnum.TENANT) {
+      passed = this.processFilterEquality(
+        filter,
+        filterProcessingDetails,
+        variables
+      );
+    }
+
+    if (filter.on === FilterPartTypeEnum.WEBHOOK) {
+      if (process.env.NODE_ENV === 'test') return true;
+
+      const webhookPayload = await this.buildPayload(variables, properties);
+      const res = await this.getWebhookResponse(
+        filter,
+        variables,
+        webhookPayload,
+        properties
+      );
+      passed = this.processFilterEquality(filter, filterProcessingDetails, {
+        payload: undefined,
+        webhook: res,
+      });
+    }
+
+    if (filter.on === FilterPartTypeEnum.PAYLOAD) {
+      passed = this.processFilterEquality(
+        filter,
+        filterProcessingDetails,
+        variables
+      );
+    }
+
+    if (filter.on === FilterPartTypeEnum.SUBSCRIBER) {
+      passed = this.processFilterEquality(
+        filter,
+        filterProcessingDetails,
+        filterData
+      );
+    }
```

Comment:
As I understand this filter is based on the custom properties stored in the subscriber entity instead of processing the filter based on the variables of the event trigger payload, we need to do it in the stored subscriber entity that is being retrieved by `this.getFilterData` earlier on and stored in `filterData`.